### DISCUSSION TITLE
the session never expires

### DISCUSSION THREAD

**Discussion by GeorgeGeorgeGe** [rejetto/hfs#219] [Q&A] [answered]
Hello everybody,

Is there any way to disconnect the session to the server after duration of inactivity on the user browser, the session never expires and this increase the security risk.
Please advice.

Thanks

**Comment by rejetto** [comment]
at the moment the session is refreshed automatically if the page is open and running/powered.
Do you need it to be refreshed only when user interacts with the server? (this doesn't account for simply moving the mouse)

**Reply by GeorgeGeorgeGe** [reply]
I need the session to get disconnected after a period of inactivity from the end user ( for example when user forgets to logout), because leaving the session active all the time means that any other users on that PC can access my server without login.
if that is not possible at the moment, please tell me how to make the session be refreshed only when user interacts with the server.
Thanks.

**Comment by rejetto** [comment]
"forget to logout" is not clarifying if the page is left open or not.
If the page is closed, the session will expire in 24h hours.
This is the default time, but can be set with an environment variable, see https://github.com/rejetto/hfs#hidden-features

**Reply by GeorgeGeorgeGe** [reply]
Yes i meant by "forget to logout" that the page left open and the user is working on another tab in the browser, in this case i think the session will keep alive until the PC is turned off, correct ?
Please advice.
Thanks

**Comment by rejetto** [comment]
yes, because the page works even you are not watching it.
In this case it will expire 1 day AFTER you turn off the PC.
If you need the session to expire while the page is left open, I can add an option for that.

**Comment by GeorgeGeorgeGe** [comment]
Yes Please it would be great and more secure if you can make the session expires while the page is let open.
Regards

**Comment by rejetto** [comment]
it's done.
you'll get it in next release
<img width="427" alt="image" src="https://user-images.githubusercontent.com/1367199/233130626-de5406de-f380-4e51-92be-4c145dd87d27.png">

**Reply by GeorgeGeorgeGe** [reply]
That would be GREAT !

**Reply by rejetto** [reply]
find this feature in 0.44 beta
https://github.com/rejetto/hfs/discussions/221

**Reply by GeorgeGeorgeGe** [reply]
Perfect.
what is the default value of  SESSION_DURATION in the new version please?

**Reply by rejetto** [reply]
it didn't change, 24 hours

**Reply by GeorgeGeorgeGe** [reply]
so if you turn off the option "keep you logged in while the page is left open and the computer is on" what would be changed ?

**Reply by rejetto** [reply]
enabled=automatic session refresh
disabled=session refresh only with user interaction

24 hours is from last session refresh

**Reply by GeorgeGeorgeGe** [reply]
Perfect !
another suggestion please;
is there any way to automatically log off the front end user after a duration of inactivity of this application web pages? so he should sign in again to continue ? like when gmail or yahoo disconnect us after a period of inactivity on their web site.
Thanks

**Reply by rejetto** [reply, marked as answer]
what you say is exactly what the new option is for.
just disable it!

**Reply by GeorgeGeorgeGe** [reply]
I disable it , it is still not working !!!
i am still able to connect to the server after few hours on inactivity without signing in !!
Please advise.

**Reply by rejetto** [reply]
saying "few hours" is not enough, the default timeout is 24 hours, so you should specify how many hours you waited and if you changed the timeout or left it to 24

**Reply by GeorgeGeorgeGe** [reply]
so what is the difference between the SESSION_DURATION ( Default 24 Hours) and the option that you have just added (Keep session alive- disabled) ?

**Reply by GeorgeGeorgeGe** [reply]
How can i change the SESSION_DURATION Please ?

**Reply by rejetto** [reply]
Google how to set an environment variable on your operating system

**Comment by rejetto** [comment]
are you using the .exe distribution, on Windows?

**Comment by GeorgeGeorgeGe** [comment]
Yes I use the exe distribution, on Windows.

**Reply by rejetto** [reply]
and do you need it to start at boot?

**Reply by GeorgeGeorgeGe** [reply]
Yes Please

**Reply by rejetto** [reply]
and did you manage to make it start at boot?

**Reply by GeorgeGeorgeGe** [reply]
Not Yet

**Reply by GeorgeGeorgeGe** [reply]
when i use the select screen and check the checkbox beside the filter text box to select the whole list, it checks only the first checkbox in the list and leave all others unchecked although you have a text saying that all items selected ( on the right side of  the filter text box).
Please advise.
Thanks

**Reply by rejetto** [reply]
please, open an "issue" about that instead of replying here
https://github.com/rejetto/hfs/issues

**Comment by gogi100** [comment]
in my case, i have many users. they upload documents then they does not log out, the session stays active but they do not nothing. i need automatically  disconnect after 1 hour

**Reply by rejetto** [reply]
Then set environment variable as stated above.
1 hour is 3600 seconds.

**Reply by gogi100** [reply]
the environment variable i set on server where is the hfs or workstation?

**Reply by rejetto** [reply]
Env must be set on server

**Comment by gogi100** [comment]
what i set in the yaml file for Keep session alive disabled? every time when i disable Keep session alive in admin-panel after refreshing i receive status on?

**Reply by rejetto** [reply]
I just tried on my server and it stays disabled even after reload. Are you sure?

**Comment by gogi100** [comment]
Yes,i sure

**Comment by rejetto** [comment]
i've managed to make you configure the session duration easily, on the interface
<img width="251" alt="image" src="https://github.com/rejetto/hfs/assets/1367199/02541342-3f13-4a74-8279-eafaf5ce055f">

I may decide to remove the old "env" method in the future, but for now both are supported, to avoid disrupting your configuration.
Find this in latest beta #241 or next release.
